### PR TITLE
Style deal component

### PR DESCRIPTION
**Pull request by brkidder**
@cadecameron @d-rowe 

Major styling in deal.jsx
Added Price in an attempt to help declutter Price. Suggestions welcome.
Minor code added. Console.logs are purposeful for now.

Thank you for your time!

### CODE REVIEW

File: client/src/components/checkout.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import $ from 'jquery';
 import Deals from './deals';
 
-class App extends React.Component {
+class Checkout extends React.Component {
```
Comment (cadecameron) [inline client/src/components/checkout.jsx]: I think the rename makes sense. 👍

File: client/src/components/deal.jsx [hunk 2]
```diff
@@ -1,29 +1,198 @@
...
-      <span>
-        {' $' + deal.salePrice}
-      </span>
-    </p>
-    <p>
-      <span>{deal.bought + '+ bought '}</span>
-      <span>{deal.salePercent + '% OFF'}</span>
-    </p>
-  </label>
+    {onSale ? (
```
Comment (cadecameron) [inline client/src/components/deal.jsx]: Nice use of conditionals!

File: client/src/components/deal.jsx [hunk 3]
```diff
@@ -1,29 +1,198 @@
...
+  letter-spacing: 0px;
+  padding: 0px;
+  margin: 0px;
+  top: 14px;
+  left: 2px;
+`;
+
+const PercentWrap = styled(DealWrap)`
+  color: ${props => props.onSale ? "rgb(255, 0, 0)" : "rgb(52, 135, 0)"};
+  background-color: ${props => props.onSale ? "rgb(255, 210, 210)" : "rgb(234, 252, 222)"};
```
Comment (cadecameron) [inline client/src/components/deal.jsx]: Did you manage to get this working in the end?

Comment (brkidder) [reply client/src/components/deal.jsx]: i had the red color coded rgb(255,0,0,0) i had too many 0's

File: client/src/components/checkout.jsx (hunk 1 above)
Comment (brkidder) [reply client/src/components/checkout.jsx]: Yeah App was a bit too general

File: client/src/components/deals.jsx [hunk 4]
```diff
@@ -1,19 +1,21 @@
 import React from 'react';
 import Deal from './deal';
 
-const Deals = (props) => (
-  props.deals.map((deal, i) => {
-    const key = i + 1;
-    return (
-      <div key={key}>
+const Deals = ({ deals, onChange, selectedOption, onSale }) => (
```
Comment (cadecameron) [inline client/src/components/deals.jsx]: 🎉

File: client/src/components/deal.jsx [hunk 5]
```diff
@@ -1,29 +1,198 @@
...
+    top: 3px;
+    left: 3px;
+    width: 11px;
+    height: 11px;
+    border-radius: 50%;
+    background: #0076d6;
+  }
+`;
+
+const PriceWrap = styled(DealWrap)`
```
Comment (d-rowe) [inline client/src/components/deal.jsx]: Nice use of style extending

Comment (cadecameron) [inline client/src/components/deal.jsx]: Nice use of style extenders 👍

File: client/src/components/deals.jsx [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
+  <form>
+    {deals.map((deal, i) => {
+      const key = i + 1;
```
Comment (cadecameron) [inline client/src/components/deals.jsx]: Is the `i + 1` necessary? Is your code relying on this key property anywhere else?

File: client/src/components/deals.jsx (hunk 4 above)
Comment (d-rowe) [inline client/src/components/deals.jsx]: Nice use of destructuring in this refactor

**Comment by d-rowe** [comment]
Good work, this refactor makes everything look a lot more manageable.

File: client/src/components/deals.jsx (hunk 6 above)
Comment (brkidder) [reply client/src/components/deals.jsx]: I just didn't want a 0 based key, but no other reason than that.

Comment (d-rowe) [reply client/src/components/deals.jsx]: Was wondering this too. Could the unique deal title be used for key?

**Comment by brkidder** [comment]
Thank you! @cadecameron @d-rowe for taking a look first thing this morning.

**Comment by brkidder** [comment]
Didn't mean to close it. Refactored the ordering of the style attributes. refactored key to something meaningful and refactored duplicate code from extending styles

**Comment by d-rowe** [comment]
Just checked out the new changes, looks great to me.

**Pull request merged**